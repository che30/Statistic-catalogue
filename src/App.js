import './App.css';

function App() {
  return (
    <div className="App">
      <p>testing </p>
    </div>
  );
}

export default App;
