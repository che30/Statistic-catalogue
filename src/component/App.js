import React from 'react';
import { SearchInput } from './SearchInput';
import Movies from './Movies';
import Navbar from './Navbar';

function App() {
  return (
    <div className="">
      <Navbar />
      <SearchInput />
      <Movies />
    </div>
  );
}

export default App;
