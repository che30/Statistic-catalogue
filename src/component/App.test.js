import React from 'react';
import renderer from 'react-test-renderer';
 import {render} from '@testing-library/react'
 import { createStore, compose, applyMiddleware } from 'redux';
 import { Provider } from 'react-redux';
 import rootReducer from '../reducers';
 import thunk from 'redux-thunk';
 import App from './App';

describe('search input methods', () => {
     const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const store = createStore(rootReducer(), composeEnhancer(applyMiddleware(thunk)));
  test('finds the text search on the screen', async () => {
    const { findByText } = render(
      <Provider store={store}>
      < App />
    </Provider>
    )
    await findByText('search');
  });
   it('renders properly', () => {
     const tree = renderer
     
       .create(<Provider store={store}>
        <App/>
      </Provider>)
       .toJSON();
     expect(tree).toMatchSnapshot();
     
   });
})

