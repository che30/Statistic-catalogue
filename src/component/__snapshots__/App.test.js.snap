// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search input methods renders properly 1`] = `
<div
  className=""
>
  <nav
    className="bg-dark text-center h2 py-3 text-danger"
  >
    MOVIE DATABASE
  </nav>
  <div
    className=" search-form "
  >
    <form
      className="d-flex w-100 justify-content-around text-center mt-3 "
    >
      <input
        className="w-25 input-form text-center"
        id="input-title"
        onChange={[Function]}
        placeholder="enter movie title"
        type="input"
        value=""
      />
      <div>
        <button
          className="bg-dark text-white px-3 rounded py-1"
          onClick={[Function]}
          type="button"
        >
          search 
        </button>
      </div>
    </form>
  </div>
  <div>
    loading movie data
  </div>
</div>
`;
